app/src/main/java/de/danoeh/antennapod/ui/home/sections/QueueExpanableSection.java

```diff
@@ -0,0 +1,37 @@
+package de.danoeh.antennapod.ui.home.sections;
+
+import androidx.fragment.app.Fragment;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.fragment.QueueFragmentInHome;
+import de.danoeh.antennapod.ui.home.HomeSection;
+
+public class QueueExpanableSection extends HomeSection {
```

Comment:
I'm a bit confused by this code. Can't we just show `QueueFragmentInHome` directly by `HomeFragment`, without having to use a dummy section?

Reply:
Yes, but i feel its cleaner like this, so we have the TAGs and the isExpandable which makes it easier for settings and the save preferences. Plus the logic of expanding is mostly neatly in the section so home fragment just cares about displaying the section. You can look at the git history, I had it without the sections and there was some vquirky logic in home fragment and not nice reflection.

Reply:
The expandable section is not really a normal home section and needs to be handled differently anyway. HavinHaving classes that do nothing or return null in most methods feels more like a hack than a direct implementation. Why can't you just make two arrays (one with tags and one with names) containing the new classes (EpisodesFragementInHome etc)? Then you can just add the corresponding Fragment with a similar switch statement as the home sections, just for the bottom half. No need to create dummy classes just to add the entries to the same switch statement

Reply:
Hmm guess I can retractor again, but I'm traveling rn and will get to it in April earliest

Reply:
sooo loked into it again, and would suggest du keep it like this, not using sections would mean the **section title layout** must be recreated which seems like a bad way of duplicate code and would make adjustements to the section layout more complicated.

Reply:
While it would duplicate the title of the home section, I think this is still the better option. The variant implemented right now adds both a RecyclerView and a fragment container to every single home section and then needs to deal with hiding the unused one. A lot of the code in the home section then need case distinctions for whether this is a normal home section or an expanded one. The expanded ones return `null` or dummy values in their methods, which causes quite a bit of "dead code" that makes things harder to understand for new contributors. Finally, these dummy sections are fragments that do basically nothing but showing a fragment themselves, which is a bit of unnecessary nesting.

If you think duplicating the title section is a problem, it could also be moved to its own XML file and included from the sections and expandable sections.

Bottom sections are a different thing using different layout, settings and setup code. Integrating this into normal home sections and then having case distinctions everywhere makes the code more complex than it needs to be.